### DISCUSSION TITLE
Is anti-phishing a good idea?

### DISCUSSION THREAD

**Discussion by datamweb** [codeigniter4/shield#229] [Ideas]
Hello friends.
I see a lot of people in the news who have been **harmed by phishing**.
They enter the username, password without knowing that the **site is fake** and are thus attacked.
When I read [How to Spot a Phishing Website ](https://www.wildfirecu.org/education-and-resources/blog/blog-post/wildfire-blog/how-to-spot-a-phishing-website). I was interested in adding a **feature to the Shield**.
Anti-phishing, which allows the user to carefully check the site address **once** and define a **seal** for himself.
After that, each time you visit the site, if you see the **same seal**, you will continue the login process.
@lonnieezell , @kenjis , @MGatner I'm very interested in adding this to **Shield**, but I do not know if it is useful to you.
If you think this is useful for Shield, please let me know so I can send a **PR**.

**Comment by MGatner** [comment]
It would be cool to have. Whether it is part of Shield or its own extension package is up for discussion. I think there are lots of different ways we might want to expand the security offerings and I would be in favor of separating these out to make maintenance easier and keep this library from bloating.

**Reply by vimkaf** [reply]
I agree with you. In my opinion Shield should be a pack for authentication and authorization, @datamweb  anti-phising is a very fantastic idea but i believe it should be an addon to the Shield package.

**Reply by datamweb** [reply]
> It would be cool to have. Whether it is part of Shield or its own extension package is up for discussion. I think there are lots of different ways we might want to expand the security offerings and I would be in favor of separating these out to make maintenance easier and keep this library from bloating.

I realized, Thank you for your answer.

**Reply by datamweb** [reply]
> I agree with you. In my opinion Shield should be a pack for authentication and authorization, @datamweb anti-phising is a very fantastic idea but i believe it should be an addon to the Shield package.

My impression of the feedback is to provide a package for Shield.
So I will present it in a separate package.
Thanks for your feedback.

**Comment by lonnieezell** [comment]
Sorry I'm late but I agree with the others. It's a great idea and would be an awesome contribution to the ecosystem, but I don't think it's right for Shield itself. I would also say the same thing about some other "Actions" I've got in mind. I'm glad to hear you're willing to create a package, though! Can't wait to see it.

**Reply by datamweb** [reply]
Thanks **lonnieezell** . I received feedback.
Collective opinion is very important.
So I agree with you.

**Thank you for your support.**
